webapp-client/src/component/CategoryList.js

```diff
@@ -1,30 +1,74 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import ErrorMessage from './ErrorMessage';
 
 export default class CategoryList extends React.Component {
-  constructor(props){
+  constructor(props) {
     super(props);
+    this.state = {
+      hasError: false,
+      categoryList: [],
+    };
     this.addAElement = this.addAElement.bind(this);
+    this.clearHasError = this.clearHasError.bind(this);
   }
 
-  addAElement(category, index){
+  async componentDidMount() {
+    const header = { 'Content-Type': 'application/json' };
+    try {
+      const response = await fetch(
+        process.env.REACT_APP_API_URL +
+          '/category/getCategories/' +
+          this.props.userID,
```

Comment:
Ik this is probably done by formatter, but this might be easier to read on one line?

Reply:
she's done by prettier so imma leave it for now, and if our hosts suggest putting it on one line I'll change it?

Reply:
Feel free to leave this to the linter (I thought you aligned on linters so shouldn't have this type of diffs..)